errcheck/config/internal/legacy/config.go

```diff
@@ -37,15 +40,40 @@ func UpgradeConfig(cfgBytes []byte) ([]byte, error) {
 	}
 
 	var v0Cfg v0.Config
-	if legacyCfg.Args[0] != "-ignore" {
-		return nil, errors.Errorf(`errcheck-asset only supports legacy configuration if the first element in "args" is "-ignore"`)
-	}
-	for _, currIgnore := range legacyCfg.Args[1:] {
-		v0Cfg.Ignore = append(v0Cfg.Ignore, currIgnore)
+
+	args := legacyCfg.Args
+	for i := 0; i < len(args); i++ {
+		switch args[i] {
+		case "-ignore":
+			i++
+			if i >= len(args) {
+				break
+			}
+
+			v0Cfg.Ignore = append(v0Cfg.Ignore, args[i])
+		case "-exclude":
+			i++
+			if i >= len(args) {
+				break
+			}
+
+			d, err := ioutil.ReadFile(args[i])
```

Comment:
I don't think this works when the program's working directory is not the project directory (which is a possibility)

Reply:
Is there a utility or method I can use to get the project directory here?

Reply:
Unfortunately not -- the upgrade API is intentionally pretty limited and just expects to be provided with input bytes and get the upgraded input bytes in return. In general, it's not expected to be able to read other config files or project files.

Do you have a sense for the scope of the use of the flag in legacy configuration? If it's 1-5 projects, then it might make sense to require manual migration.

If its 5+ projects, then I think the automated migration is worth it. In that scenario, even though it's not 100% conceptually "correct", we can probably leave this code as-is with the expectation that config upgrade will generally be invoked from the proper location (root directory of project).

Reply:
It's only 1 or 2 projects that I know of that haven't already been converted manually. I can remove this and leave it as it was before, failing if any non-`-ignore` flags are set.

Reply:
Cool, let's just do that for simplicity's sake